interfaces/policy/helpers.go

```diff
@@ -44,6 +45,27 @@ func checkSnapType(snapType snap.Type, types []string) error {
 	return fmt.Errorf("snap type does not match")
 }
 
+func checkID(kind, id string, ids []string, special map[string]string) error {
+	if len(ids) == 0 {
+		return nil
+	}
+	if id == "" { // unset values never match
+		return fmt.Errorf("%s does not match", kind)
+	}
+	for _, cand := range ids {
+		if strings.HasPrefix(cand, "$") {
+			cand = special[cand]
+			if cand == "" {
+				continue
+			}
```

Comment:
@jdstrand also to be clear this means that they ideally need to be spotted by tooling elsewhere